### PR TITLE
Hello world example

### PR DESCRIPTION
**Pull request by dave3d**

### CODE REVIEW

File: Examples/HelloWorld/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (zivy) [inline Examples/HelloWorld/CMakeLists.txt]: Add all of the examples as tests, similar to the SimpleGaussian example [CMakeLists file](https://github.com/SimpleITK/SimpleITK/blob/next/Examples/SimpleGaussian/CMakeLists.txt).

File: Examples/CMakeLists.txt [hunk 2]
```diff
@@ -15,6 +15,11 @@ if(NOT CMAKE_PROJECT_NAME STREQUAL "SimpleITK")
 
 endif()
 
+option(SimpleITK_SHOW_TESTING_ENABLED "Enable the Show function in SimpleITK testing." OFF)
+if (NOT SimpleITK_SHOW_TESTING_ENABLED)
+  set(ENV{SITK_NOSHOW} TRUE)
```
Comment (blowekamp) [inline Examples/CMakeLists.txt]: I am not sure what this code block does. The tests are not effected by changes to when the CMake process is running.

File: Examples/HelloWorld/CMakeLists.txt [hunk 3]
```diff
@@ -6,3 +6,45 @@ target_link_libraries ( HelloWorld ${SimpleITK_LIBRARIES} )
...
+  return ()
+endif ()
+
+if ( SimpleITK_SHOW_TESTING_ENABLED )
+
+  sitk_add_test( NAME CXX.Example.HelloWorld
+    COMMAND $<TARGET_FILE:HelloWorld>
+    )
+
+  if ( WRAP_CSHARP )
```
Comment (blowekamp) [inline Examples/HelloWorld/CMakeLists.txt]: This `WRAP_CSHARP` should not be needed the `sitk_add_csharp_test` macro should take care of it.

File: Examples/HelloWorld/CMakeLists.txt [hunk 4]
```diff
@@ -6,3 +6,45 @@ target_link_libraries ( HelloWorld ${SimpleITK_LIBRARIES} )
 if ( WRAP_CSHARP )
   csharp_add_executable( HelloWorldCS SimpleITKCSharpManaged.dll ${CMAKE_CURRENT_SOURCE_DIR}/HelloWorld.cs )
```
Comment (blowekamp) [inline Examples/HelloWorld/CMakeLists.txt]: Using the `sitk_add_csharp_test macro below should take care of building the csharp code.

Comment (dave3d) [inline Examples/HelloWorld/CMakeLists.txt]: I'm not using the sitk_add_charp_test.  It's actually using sitk_add_test.  I wanted the Csharp to compile even without testing.

File: Examples/CMakeLists.txt (hunk 2 above)
Comment (dave3d) [inline Examples/CMakeLists.txt]: I wanted to force the environment variable SITK_NOSHOW to be true by default when running the tests.  Not sure it actually accomplishes that though.

Comment (blowekamp) [inline Examples/CMakeLists.txt]: The `ENVIRONMENT` property needs to be modified for each test. Just about every test goes through the sitk_add_test macro. The code block from the python and lua add_tests can be consolidated in to sitk_add_test. The `[APPEND`](https://cmake.org/cmake/help/v3.0/command/set_property.html) option should be added to the `set_property` command as well.

File: Examples/HelloWorld/CMakeLists.txt (hunk 4 above)
Comment (blowekamp) [inline Examples/HelloWorld/CMakeLists.txt]: Using `sitk_add_csharp` would make it consistent with the other two csharp examples and java. the C++ does appears to always get compiled though.

I am not sure the behavior of `BUILD_TESTING` is consistent. The behavior could be make consistent inside the `sitk_add_java_test` and `sitk_add_csharp_test`. Please remove the separate sharp compilation, and use the macro. The logic for building the tests can be modified in another patch inside these  testing macros.

**Comment by blowekamp** [comment]
Github seems to be a little confused on this patch now after the rebase. I am going to close it so you can open a new pull request, that we can review. 

Thanks Brad

**Pull request closed without merging**